case-summarization/deploy/case-summarization.yaml

```diff
@@ -0,0 +1,450 @@
+    sa-east-1:      {BucketName: aws-managed-cost-intelligence-dashboards-sa-east-1}
+    us-east-1:      {BucketName: aws-managed-cost-intelligence-dashboards-us-east-1}
+    us-east-2:      {BucketName: aws-managed-cost-intelligence-dashboards-us-east-2}
+    us-west-1:      {BucketName: aws-managed-cost-intelligence-dashboards-us-west-1}
+    us-west-2:      {BucketName: aws-managed-cost-intelligence-dashboards-us-west-2}
+    #todo: add af-south-1
+    #todo: add ap-southeast-3
+    #todo: add eu-south-1
+    #todo: add eu-central-2
+
+Resources:
+
+  LlmLambdaLayer:
+    Type: AWS::Lambda::LayerVersion
+    Properties:
+      LayerName: !Sub "${ResourcePrefix}llm-layer"
+      Description: Layer for Support Case Summarization Lambda
+      Content:
+        S3Bucket: !If
+          - LambdaLayerBucketPrefixIsManaged
+          - !FindInMap [RegionMap, !Ref 'AWS::Region', BucketName]
+          - !Sub '${LambdaLayerBucketPrefix}-${AWS::Region}'
+        S3Key: 'cid-llm-lambda-layer/llm-0.0.1.zip'
+      CompatibleArchitectures:
+        - x86_64
+      CompatibleRuntimes:
+        - python3.10
+        - python3.11
+        - python3.12
+        - python3.13
+
+  SummarizationLambdaRole:
+    Type: AWS::IAM::Role
+    Properties:
+      RoleName: !Sub "${ResourcePrefix}${CFDataName}-LambdaRole"
+      AssumeRolePolicyDocument:
+        Statement:
+          - Action:
+              - sts:AssumeRole
+            Effect: Allow
+            Principal:
+              Service:
+                - !Sub "lambda.${AWS::URLSuffix}"
+        Version: 2012-10-17
+      ManagedPolicyArns:
+        - !Sub "arn:${AWS::Partition}:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole"
+      Path: /
+      Policies:
+        - PolicyName: "S3-Access"
+          PolicyDocument:
+            Version: "2012-10-17"
+            Statement:
+              - Effect: "Allow"
+                Action:
+                  - "s3:PutObject"
+                  - "s3:GetObject"
+                  - "s3:PutObjectAcl"
+                Resource:
+                  - Fn::Sub:
+                      - "arn:${AWS::Partition}:s3:::${DestinationBucket}/support-cases/*"
+                      - DestinationBucket: {"Fn::ImportValue": "cid-DataCollection-Bucket"}
+              - Effect: "Allow"
+                Action:
+                  - "s3:ListBucket"
+                Resource:
+                  - Fn::Sub:
+                      - "arn:${AWS::Partition}:s3:::${DestinationBucket}"
+                      - DestinationBucket: {"Fn::ImportValue": "cid-DataCollection-Bucket"}
+        - PolicyName: "Bedrock-Access"
+          PolicyDocument:
+            Version: "2012-10-17"
+            Statement:
+              - Effect: "Allow"
+                Action:
+                  - "bedrock:ListFoundationModels"
+                  - "bedrock:InvokeModel"
+                Resource: "*" # cannot restrict this
```

Comment:
I know this role is only going to lambda and it's using direct models, but if customers wanted to track their spend on Bedrock for example they could define which model to use and then a application profile could be created from it with relevant tags to ensure they have visibility of model invocations.

Otherwise, a custom resource could help here should further restriction be required from security point of view, doing the same resolution you have in place and generating the appropriate policy.